app/models/concerns/assignable.rb

```diff
@@ -4,12 +4,20 @@ module Assignable
   extend ActiveSupport::Concern
 
   included do
+    include Canaid::Helpers::PermissionsHelper
+
     has_many :user_assignments, as: :assignable, dependent: :destroy
 
     default_scope { includes(user_assignments: :user_role) }
 
+    scope :readable_by_user, lambda { |user|
+      joins(user_assignments: :user_role)
+        .where(user_assignments: { user: user })
+        .where('? = ANY(user_roles.permissions)', "::#{self.class.to_s.split('::').first}Permissions".constantize::READ)
```

Comment:
won't work below MyModule

Reply:
It should still be applicable for all Assignables, for below MyModule we might need custom scopes on those models anyway, depeding on implementation.